### DISCUSSION TITLE
Welcome to Subfish Discussions!

### DISCUSSION THREAD

**Discussion by SethRobinson** [SethRobinson/Subfish#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Nokhodi** [comment]
Hi, Each time you click on "Find", it crashes and closes immediately.
in terms of features, sometimes you want to search for a set of similar words, such as (Ukraine/ukraine, Ukrainian/ukrainian, Ukrainians/ukrainians, Slavs/slavs, Slav/slav, etc ...) not just a single word or phrase at a time, and you also want to set conditions for Subfish that:
1. Only search for videos that contain all of these words.
2. Search for videos that contain at least one of these words or any video in which these words are used.
You can put a checkbox for these commands.
Also, if while searching and downloading subtitles from videos on a channel, the internet connection is suddenly lost or the VPN IP is changed, the subtitle download process will stop and you will have to close the software or start over, so you will lose progress. And you have to wait again to get to the previous point, I wish there was a feature that you could save the progress or the software could scan the download folder so that the downloaded subtitles would not be replaced again and the progress would continue from where it left off. 
I believe that if you add these features to Subfish, it will make it more flexible and usable, and therefore more popular.

**Reply by SethRobinson** [reply]
Hi, did you see my answer on your bug thread?  https://github.com/SethRobinson/Subfish/issues/2

About searching click the "?" to bring up some regex examples that help with trickier searches.  Unfortunately regex is the most complicated and evil thing on earth so I agree a big dialog box with simple options would be much better for searching.  If you're brave though: https://en.wikipedia.org/wiki/Regular_expression

On being forced to re-download subs, I can't remember the specifics of the limitations of youtube-dl (the program actually doing the downloading in the background)  but I don't think it can start downloading from the middle of a playlist so a crash really does ruin things, might not be something I can fix without digging into the youtube-dl source.  (although I just switched to the yt-dlp variant of it, so it's possible the situation is changed now, hrm. 

You can get a reverse playlist link from youtube which might help a bit. (can start downloading from the other side) 

I generally only add the things I personally need unless it's quick and easy but hopefully another programmer will come along and add more stuff someday. I always want to fix any crashes/bugs ASAP though, just need a way to reproduce.
